src/main/java/com/google/devtools/build/lib/rules/android/DexArchiveAspect.java

```diff
@@ -182,8 +183,12 @@ public AspectDefinition getDefinition(AspectParameters params) {
                     // For proto_lang_toolchain rules, where we just want to get at their runtime
                     // deps.
                     ImmutableSet.of(ProtoLangToolchainProvider.class)))
-            .addRequiredToolchains(
-                Label.parseAbsoluteUnchecked(toolsRepository + sdkToolchainLabel))
+            .addToolchainType(
+                ToolchainTypeRequirement.builder()
+                    .toolchainType(
+                        Label.parseAbsoluteUnchecked(toolsRepository + sdkToolchainLabel))
+                    .mandatory(true)
```

Comment:
I'm a little unhappy with the verbosity here. Should I add a shorter version?

One thought:
```
.addToolchainType(ToolchainTypeRequirement.builder(typeLabel))
```

Thoughts?

Reply:
Sure. As currently written I had some trouble identifying when the builder resolves and how it generally fits with the pattern.

Reply:
Okay, added them as #14825, once that's in I'll rebase this.

Reply:
Rebased.